extensions/azure/storage/AzureBlobStorage.cpp

```diff
@@ -74,4 +75,14 @@ bool AzureBlobStorage::deleteBlob(const DeleteAzureBlobStorageParameters& params
   }
 }
 
+std::optional<uint64_t> AzureBlobStorage::fetchBlob(const FetchAzureBlobStorageParameters& params, io::BaseStream& stream) {
+  try {
+    auto fetch_res = blob_storage_client_->fetchBlob(params);
+    return internal::pipe(fetch_res.get(), &stream);
```

Comment:
What do you think about returning an `InputStream` instead of piping its contents into an `OutputStream`? The caller could still do a piping if they wanted, but they may also be able to use the `InputStream` in a more efficient way.

Reply:
I'm not sure if it's worth generalizing it at the moment as we only have one use case for this operation, and I don't think we will have any other in the future. If another use case arises I think it can be changed in that scope.